app/controllers/requests_controller.rb

```diff
@@ -125,6 +125,12 @@ def unsuccessful_action(format, method)
     format.json { render json: @request.errors, status: :unprocessable_entity }
   end
 
+  def prepare_params
```

Comment:
I do like that this is now it's own method
However, you should explicitly check whether params[:request][:name], etc. exists before accessing it.
If they don't at the moment the server will probably error out